server/server.js

```diff
@@ -22,12 +22,165 @@ app.use(express.static(reactStaticDir));
 app.use(express.static(uploadsStaticDir));
 app.use(express.json());
 
-app.get('/api/hello', (req, res) => {
-  res.json({ message: 'Hello World!' });
+app.get('/api/cart', async (req, res) => {
+  try {
+    const sql = `
+    select *
+      from "cart"
+    `;
+    const results = await db.query(sql);
+    res.json(results.rows);
+    // console.log(results.rows);
+  } catch (err) {
+    res.status(500).json({ error: 'an unexpected error occurred' });
+  }
+});
+
+app.get('/api/wishlist', async (req, res) => {
+  try {
+    const sql = `
+    select *
+      from "wishlist"
+    `;
+    const results = await db.query(sql);
+    res.json(results.rows);
+  } catch (error) {
+    res.status(500).json({ error: 'an unexpected error occurred' });
+  }
+});
+
+app.post('/api/wishlist', async (req, res) => {
+  try {
+    // might have issues with description
+    const { title, author, isbn, rating, image, description, price } = req.body;
+    if (!title || !author || !isbn || !image || !description || !price) {
+      return res.status(400).json({ error: 'title, author, isbn, description, image, and price are required' });
```

Comment:
I find it easier to use the error middleware approach. It's actually installed (on line 148) so you'd just as well use it!